subsys/net/openthread/Kconfig.defconfig

```diff
@@ -251,6 +251,15 @@ endif # OPENTHREAD_CSL_RECEIVER
 
 endif # !OPENTHREAD_THREAD_VERSION_1_1
 
+if OPENTHREAD_THREAD_VERSION_1_4
+
+# Thread 1.4 dependencies
+config OPENTHREAD_TCP_ENABLE
+	bool
+	default y
+
+endif # OPENTHREAD_THREAD_VERSION_1_4
+
 if OPENTHREAD_COPROCESSOR_RCP
 
 config MBEDTLS_PSA_KEY_SLOT_COUNT
```

Comment:
No to the below too, remove types from all symbols that already exist

Reply:
This is not in the scope of this PR. I can create a follow-up PR removing types from symbols that already exist.

Reply:
addressed in https://github.com/nrfconnect/sdk-nrf/pull/17707